frontend/wallet/src/render/views/settings/DelegationSettings.re

```diff
@@ -0,0 +1,188 @@
+      </a> 
+      <Spacer width=0.2/>
+      <AccountName pubkey=publicKey className=Styles.breadcrumbText/>
+    </div>
+         {switch (result) {
+          | NotCalled
+          | Loading => React.null
+          | Error(err) => <Alert kind=`Danger message=err##message />
+          | Data(_) =>
+            goBack();
+            React.null;
+          }}
+         <div className=Theme.Text.Header.h3>
+           {React.string("Delegate Participation To")}
+         </div>
+         <Spacer height=1. />
+         <TextField
+           label="Key"
+           value={
+             Option.map(~f=PublicKey.toString, state.delegate)
+             |> Option.withDefault(~default="")
+           }
+           mono=true
+           onChange={value =>
+             changeState(_ =>
+               {
+                 delegate: Some(PublicKey.ofStringExn(value)),
+                 fee: state.fee,
+               }
+             )
+           }
+         />
+         <Spacer height=1. />
+         <div>
+           <div className=Styles.label>
+             {React.string("Transaction fee")}
+           </div>
+           <ToggleButton
+             options=[|"Standard: 5 Coda", "Custom Amount"|]
+             selected=feeSelectedValue
+             onChange=onChangeFee
+           />
+           {switch (state.fee) {
+            | DefaultAmount => React.null
+            | Custom(feeAmount) =>
+              <>
+                <Spacer height=1. />
+                <TextField.Currency
+                  label="Fee"
+                  value={
+                    feeAmount == Int64.zero ? "" : Int64.to_string(feeAmount)
+                  }
+                  placeholder="0"
+                  onChange={value => {
+                    let serializedValue =
+                      switch (value) {
+                      | "" => Int64.zero
+                      | nonEmpty => Int64.of_string(nonEmpty)
+                      };
+                    changeState(_ =>
+                      {
+                        delegate: state.delegate,
+                        fee: Custom(serializedValue),
+                      }
+                    );
+                  }}
+                />
+              </>
+            }}
+         </div>
+         <Spacer height=2. />
+         <span
+           className=Css.(
+             merge([
+               Theme.Text.Body.regular,
+               style([color(Theme.Colors.slate)]),
+             ])
+           )>
+           {React.string("Delegation will take effect in 24-36 hours.")}
+         </span>
+         <Spacer height=1. />
+         <div className=Css.(style([display(`flex)]))>
+           <Button label="Cancel" style=Button.Gray onClick={_ => goBack()} />
+           <Spacer width=1. />
+           <Button
+             label="Delegate"
+             style=Button.Green
+             disabled=loading
+             onClick={_ => mutate(~variables, ~refetchQueries=[|"getAccountInfo"|], ()) |> ignore}
```

Comment:
Might want to also refetch queryDelegation to change the header text

Reply:
👍 got it!